### DISCUSSION TITLE
Re-formatting file?

### DISCUSSION THREAD

**Discussion by superDuperCyberTechno** [helix-editor/helix#9043] [General] [answered]
I just copy/pasted some JS code into its own new dedicated .js file and the formatting is all off since it comes from inline HTML.

Isn't Helix supposed to reformat on save? How do I force re-formatting?  `=` is disabled as per the docs, why?

**Comment by dead10ck** [comment]
Only a few languages have auto format on save by default; for most, you have to turn it on in your `languages.toml`. Auto formatting is a fairly invasive thing to do for many projects, so it's only turned on in ecosystems which have a very well established and accepted standard formatter, such as Rust and Go.

But you can format manually any time as well by just typing `:format`.

**Comment by superDuperCyberTechno** [comment]
But `:format` does nothing?

I installed from the Arch repos: https://archlinux.org/packages/extra/x86_64/helix/

**Reply by dead10ck** [reply]
Formatting is done by the configured formatter or language server, so if nothing happens, then check out your LS.

**Comment by superDuperCyberTechno** [comment, marked as answer]
Alright, sorry for my ignorance. I didn't realize that I needed to install the language servers seperately.

Installing `typescript-language-server` did the trick.
